src/providers.ts

```diff
@@ -1,16 +1,18 @@
 import type * as monaco from 'monaco-editor';
 type Monaco = typeof monaco;
-import type {IGrammar, IRawGrammar, IRawTheme, IOnigLib, StackElement} from 'vscode-textmate';
+import type {IGrammar, IRawGrammar, IRawTheme, IOnigLib, StateStack} from 'vscode-textmate';
 import type {LanguageId, LanguageInfo} from './register';
 
 import {INITIAL, Registry, parseRawGrammar} from 'vscode-textmate';
 // @ts-ignore
-import {generateTokensCSSForColorMap} from 'monaco-editor/esm/vs/editor/common/modes/supports/tokenization.js';
+import {generateTokensCSSForColorMap} from 'monaco-editor/esm/vs/editor/common/languages/supports/tokenization.js';
 // @ts-ignore
-import {TokenizationRegistry} from 'monaco-editor/esm/vs/editor/common/modes.js';
+import {TokenizationRegistry} from 'monaco-editor/esm/vs/editor/common/tokenizationRegistry.js';
```

Comment:
I just went through the same exercise... the correct update here is this:

```suggestion
import {TokenizationRegistry} from 'monaco-editor/esm/vs/editor/common/languages.js';
```